faasmcli/faasmcli/tasks/knative.py

```diff
@@ -238,15 +238,40 @@ def _deploy_knative_fn(
 
 @task
 def install(ctx):
+    """
+    Install knative on an existing k8s cluster
+    """
     specs = [
-        "serving-crds.yaml",
-        "serving-core.yaml",
-        "serving-istio.yaml",
+        ["knative/serving", "serving-crds.yaml"],
+        ["knative/serving", "serving-core.yaml"],
+        ["knative/net-istio", "istio.yaml"],
+        ["knative/net-istio", "net-istio.yaml"],
+        ["knative/serving", "serving-default-domain.yaml"],
     ]
 
     for s in specs:
         _kubectl_apply(
-            "https://github.com/knative/serving/releases/download/v{}/{}".format(
-                KNATIVE_VERSION, s
+            "https://github.com/{}/releases/download/v{}/{}".format(
+                s[0], KNATIVE_VERSION, s[1]
+            )
+        )
+
+
+@task
+def uninstall(ctx):
+    """
+    Uninstall knative from an existing k8s cluster
+    """
+    specs = [
+        ["knative/net-istio", "net-istio.yaml"],
+        ["knative/net-istio", "istio.yaml"],
+        ["knative/serving", "serving-core.yaml"],
+        ["knative/serving", "serving-crds.yaml"],
+    ]
```

Comment:
This list of 4 values is duplicated, could we perhaps have a constant called `KNATIVE_SPECS` and just add `serving-default-domain.yaml` in the `install` task? Why don't we call delete on `serving-default-domain.yaml` as well?

Reply:
The reason was that the lists are not traversed in the same order, as otherwise I would sometimes get errors complaining that certain component were not there.

And the `serving-default-domain.yml` is a one-off task, and creates no resources.